src/main/java/io/confluent/idesidecar/restapi/resources/ConnectionsResource.java

```diff
@@ -83,11 +84,35 @@ public Uni<ConnectionsList> listConnections() {
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Connection createConnection(ConnectionSpec connectionSpec)
-      throws ConnectionNotFoundException, CreateConnectionException {
-
-    var newSpec = connectionStateManager.createConnectionState(connectionSpec);
-    return Connection.from(newSpec);
+  public Uni<Connection> createConnection(
+      ConnectionSpec connectionSpec,
+      @Schema(
+          description = "Whether to check the connection without creating it",
+          defaultValue = "false"
+      )
+      @QueryParam("dry_run")
+      boolean dryRun
+  ) throws ConnectionNotFoundException, CreateConnectionException {
+    if (dryRun) {
+      // Just test the connection and return the status
+      var testedState = connectionStateManager.testConnectionState(connectionSpec);
+      // Get the status of the connection
+      var futureStatus = testedState.getConnectionStatus();
+      // And create a uni that will complete when the status is available
+      return Uni
+          .createFrom()
+          .completionStage(futureStatus.toCompletionStage())
+          .map(connectionStatus ->
+              Connection.from(testedState, connectionStatus)
+          );
+    }
+    // Create a real connection state and return the connection using the initial status
+    var newState = connectionStateManager.createConnectionState(connectionSpec);
+    return Uni
+        .createFrom()
+        .item(
+            Connection.from(newState, newState.getInitialStatus())
```

Comment:
For real create-connection behavior, I changed this line to `Connection.from(newState)` to this, and removed the `Connection.from(ConnectionState state)` (without a status) to help ensure we always pass in the correct state.